patches/tModLoader/Terraria/ModLoader/Config/ConfigManager.cs

```diff
@@ -292,33 +391,29 @@ public static object AlternateCreateInstance(Type type)
 
 	// Gets an Attribute from a property or field. Attribute defined on Member has highest priority,
 	// followed by the containing data structure, followed by attribute defined on the Class.
-	public static T GetCustomAttribute<T>(PropertyFieldWrapper memberInfo, object item, object array) where T : Attribute
+	public static T? GetCustomAttribute<T>(PropertyFieldWrapper memberInfo, object? item, object? array) where T : Attribute
 	{
 		// Class
-		T attribute = (T)Attribute.GetCustomAttribute(memberInfo.Type, typeof(T), true);
+		T? attribute = (T?)Attribute.GetCustomAttribute(memberInfo.Type, typeof(T), true);
 		if (array != null) {
 			// item null?
 			//	attribute = (T)Attribute.GetCustomAttribute(item.GetType(), typeof(T), true) ?? attribute; // TODO: is this wrong?
 		}
 		// Member
-		attribute = (T)Attribute.GetCustomAttribute(memberInfo.MemberInfo, typeof(T)) ?? attribute;
+		attribute = (T?)Attribute.GetCustomAttribute(memberInfo.MemberInfo, typeof(T)) ?? attribute;
 		return attribute;
 		// TODO: allow for inheriting from parent's parent? (could get attribute from parent ConfigElement)
 	}
 
-	public static T GetCustomAttribute<T>(PropertyFieldWrapper memberInfo, Type type) where T : Attribute
+	public static T? GetCustomAttribute<T>(PropertyFieldWrapper memberInfo, Type type) where T : Attribute
 	{
-		// Class
-		T attribute = (T)Attribute.GetCustomAttribute(memberInfo.Type, typeof(T), true);
-
-		attribute = (T)Attribute.GetCustomAttribute(type, typeof(T), true) ?? attribute;
-
-		// Member
-		attribute = (T)Attribute.GetCustomAttribute(memberInfo.MemberInfo, typeof(T)) ?? attribute;
-		return attribute;
+		return
+			(T?)Attribute.GetCustomAttribute(memberInfo.MemberInfo, typeof(T)) ?? // on the member itself
+			(T?)Attribute.GetCustomAttribute(type, typeof(T), true) ?? // on a provided fallback type
+			(T?)Attribute.GetCustomAttribute(memberInfo.Type, typeof(T), true); // on the member type. TODO is there ever a reason to use this?
```

Comment:
See my todo here. I don't think this last fallback makes sense. `type` should probably be called `fallbackType`. Then you can pass in `MemberInfo` rather than `PropertyFieldWrapper`